### ISSUE TITLE
Rollback of last training epochs having zeros in P, R, mAp

### ISSUE THREAD

**Issue by cicdacc** [ultralytics/yolov5#5768]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello!

Starting from epoch 189 precision, recall and mAp suddenly fell to zeros.
I found in discussions here that it may be due to the absence of some classes in my images and labels. I do have 4 out of 50 classes that are not presented in train/val images and labels.
![zeros](https://user-images.githubusercontent.com/74823695/143221740-fa256b13-adc3-4814-a9d5-0abb81803330.png)

I'll try to add at least one sample per each of them to my train/val subsets.
But how/(should I) to rollback last epochs with zeros in order to restart training from last normal epoch (188)?

**Comment by glenn-jocher** [comment]
@cicdacc your loss jumped, likely due to an issue in your custom dataset.

**Comment by cicdacc** [comment]
Is it safe to resume training with  _python train.py --resume_  when I fix my custom dataset?

Or do I have to rollback last (bad) iterations, and if so - how to do it?

**Comment by glenn-jocher** [comment]
@cicdacc you can resume from best.pt, i.e. `python train.py --resume path/to/best.pt`

**Comment by cicdacc** [comment]
Thank you!

**Comment by mayamehta0095** [comment]
Hello,

I am facing the same issue of training epochs having zeros in P,R, mAp

Starting training for 20 epochs...


               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 1/1 [00:01<00:00,  1.52s/it]
                 all          2          0          0          0          0          0

earlier I was getting accurate result but now it shows zero.

**Comment by glenn-jocher** [comment]
@mayamehta0095 you have zero labels as your results clearly show.

**Comment by mayamehta0095** [comment]
@glenn-jocher  I have created dataset as per YOLOV5 framework and given path accurately then also I am not getting labels in training.
Can you please help me how do I resolve this?

**Comment by glenn-jocher** [comment]
@mayamehta0095 see **Train Custom Data** tutorial for correct dataset formatting:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
